include/boost/json/impl/value_ref.ipp

```diff
@@ -46,8 +46,9 @@ is_key_value_pair() const noexcept
         return false;
     auto const& e =
         *arg_.init_list_.begin();
-    if( e.what_ != what::str)
-        return false;
+    if(e.what_ != what::str &&
+        e.what_ != what::strfunc)
+            return false;
     return true;
```

Comment:
Yep this looks much better without the else, as it is absolutely clear that the function ends on line 52